libs/ledger-live-common/src/hw/index.ts

```diff
@@ -107,7 +112,45 @@ export const open = (
   for (let i = 0; i < modules.length; i++) {
     const m = modules[i];
     const p = m.open(deviceId, timeoutMs, context);
-    if (p) return p;
+    if (p) {
+      trace({
+        type: LOG_TYPE,
+        message: `Found a matching Transport: ${m.id}`,
+        context,
+        data: { timeoutMs },
+      });
+
+      if (!timeoutMs) {
+        return p;
+      }
+
+      let timer: ReturnType<typeof setTimeout> | null = null;
+
+      // Throws CantOpenDevice on timeout, otherwise returns the transport.
+      // Important: with Javascript Promise, when one promise finishes,
+      // the other will still continue, even if its return value will be discarded.
+      return Promise.race([
+        p.then(transport => {
+          // Necessary to stop the ongoing timeout
+          if (timer) {
+            clearTimeout(timer);
+          }
+
+          return transport;
+        }),
+        new Promise((_resolve, rejects) => {
```

Comment:
why the weird naming for `resolve` and `reject`?

Reply:
- `resolve` is never used, so the `_` is for the linter
- `rejects` i don't know why i added an `s` ahah, removing it